### PR TITLE
Introduce new serialize() method for Cobbler Items

### PR DESCRIPTION
**Pull request by SchoolGuy**
This is a wrapper around to_dict() to handle specific cases where you don't want specific attributes to be serialized
to the disk but you want to have properties for data and type validation.

This supersedes and fixes the problems addressed in #2724

### CODE REVIEW

**Comment by SchoolGuy** [comment]
Currently missing/known issue is that the `remote_grub_*` attributes do not accept their own format but only the ones from `remote_boot_*`.

**Comment by SchoolGuy** [comment]
I am aware that my implementation is not working in all cases but we should have covered 99 percent of cases where we do want to get a kernel or initrd. I believe that we can improve the support in a later cleanup.

File: cobbler/validate.py [hunk 1]
```diff
@@ -538,3 +541,57 @@ def validate_serial_baud_rate(baud_rate: Union[int, enums.BaudRates]) -> enums.B
     if baud_rate not in enums.BaudRates:
         raise ValueError("invalid value for serial baud Rate (%s)" % baud_rate)
     return baud_rate
+
+
+def validate_boot_remote_file(value: str) -> bool:
+    """
+    This validates if the passed value is a valid value for ``remote_boot_{kernel,initrd}``.
+
+    :param value: Must be valid URI starting with http or tftp. ftp is not supported and thus invalid.
```
Comment (nodeg) [inline cobbler/validate.py:550]:
```suggestion
    :param value: Must be a valid URI starting with http or tftp. ftp is not supported and thus invalid.
```

**Review by nodeg — APPROVED** [review]
LGTM.

**Pull request merged**